chain/client/src/stateless_validation/partial_witness/partial_witness_actor.rs

```diff
@@ -280,25 +267,19 @@ impl PartialWitnessActor {
     ) -> Result<(), Error> {
         tracing::debug!(target: "client", ?partial_witness, "Receive PartialEncodedStateWitnessMessage");
 
-        let signer = match self.my_signer.get() {
-            Some(signer) => signer,
-            None => {
-                return Err(Error::NotAValidator(format!("handle partial encoded state witness")));
-            }
-        };
+        let signer = self
+            .my_signer
+            .get()
+            .ok_or(Error::NotAValidator(format!("handle partial encoded state witness")))?;
 
-        // Validate the partial encoded state witness.
+        // Validate the partial encoded state witness and forward the part to all the chunk validators.
         if validate_partial_encoded_state_witness(
             self.epoch_manager.as_ref(),
             &partial_witness,
             &signer,
             &self.store,
         )? {
-            // Store the partial encoded state witness for self.
-            self.partial_witness_tracker
-                .store_partial_encoded_state_witness(partial_witness.clone())?;
-            // Forward the part to all the chunk validators.
-            self.forward_state_witness_part(partial_witness, &signer)?;
+            self.forward_state_witness_part(partial_witness)?;
```

Comment:
Is there any concern that we will forward the part twice (or even infinitely many times) because self-sending triggers it?

Reply:
I don't think that should be the case as we only call the forward function for handling `PartialEncodedStateWitnessMessage` (and not `PartialEncodedStateWitnessForwardMessage`).

Logically, it's a one way stream, PartialEncodedStateWitnessMessage -> forward -> PartialEncodedStateWitnessForwardMessage -> store

We can see there's only one call site.